### PR TITLE
add `r-archr`

### PR DESCRIPTION
**Pull request by mfansler**
The [`ArchR` package](https://www.archrproject.com/index.html) is routinely used in the community without a canonical Conda version. This proposes to establish one in Bioconda. The recipe is mostly derived from https://github.com/bioconda/bioconda-recipes/pull/36855.

<details>
Unfortunately, there has been minor bug fixing in the `main` branch since the last Release. In lieu of this, I propose using a commit-specific build, with the number `"8"` corresponding to the *8th commit after the v1.0.2 release* (build 0).

However, it appears the `dev` branch has been solving more substantial issues, also without version releases. As an alternative to custom build strings, perhaps we should **consider customizing the version string**? For example, something like `1.0.2_main20240226` and `1.0.3_dev20230825`, i.e., a `_{branch}{YYYYMMDD}` suffix.
</details>

> 📢 *Update 11 Oct*

Initially we can commit to the Release v1.0.2 plus a single patch from `master`. Further discussion about variants can be postponed.

<details>
A further issue to address is that packages lack upper bounds while the upstream devs frequently recommend installing older R/Bioconductor versions  (e.g., R 4.1/Bioc 3.15). It may be important to **identify incompatibilities and set upper bounds prior to merging**, in order to preempt non-functional environments.
</details>

> 📢 *Update 11 Oct*

As is frequently posted in upstream Issue responses, the release version is stable on R 4.1/Bioc 3.15. For now we propose building only for R 4.1.

CC: @jdblischak okay if I list you as a maintainer here? Any thoughts about this also welcome!

----

Please read the [guidelines for Bioconda recipes](https://bioconda.github.io/contributor/guidelines.html) before opening a pull request (PR).

### General instructions

* If this PR adds or updates a recipe, use "Add" or "Update" appropriately as the first word in its title.
* New recipes not directly relevant to the biological sciences need to be submitted to the [conda-forge channel](https://conda-forge.org/docs/) instead of Bioconda.
* PRs require reviews prior to being merged. Once your PR is passing tests and ready to be merged, please issue the `@BiocondaBot please add label` command.
* Please post questions [on Gitter](https://gitter.im/bioconda/Lobby) or ping `@bioconda/core` in a comment.

### Instructions for avoiding API, ABI, and CLI breakage issues
Conda is able to record and lock (a.k.a. pin) dependency versions used at build time of other recipes.
This way, one can avoid that expectations of a downstream recipe with regards to API, ABI, or CLI are violated by later changes in the recipe.
If not already present in the meta.yaml, make sure to specify `run_exports` (see [here](https://bioconda.github.io/contributor/linting.html#missing-run-exports) for the rationale and comprehensive explanation).
Add a `run_exports` section like this:

```yaml
build:
  run_exports:
    - ...

```

with `...` being one of:

| Case                             | run_exports statement                                               |
| -------------------------------- | ------------------------------------------------------------------- |
| semantic versioning              | `{{ pin_subpackage("myrecipe", max_pin="x") }}`     |
| semantic versioning (0.x.x)      | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}`   |
| known breakage in minor versions | `{{ pin_subpackage("myrecipe", max_pin="x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| known breakage in patch versions | `{{ pin_subpackage("myrecipe", max_pin="x.x.x") }}` (in such a case, please add a note that shortly mentions your evidence for that) |
| calendar versioning              | `{{ pin_subpackage("myrecipe", max_pin=None) }}`    |

while replacing `"myrecipe"` with either `name` if a `name|lower` variable is defined in your recipe or with the lowercase name of the package in quotes.

### Bot commands for PR management

<details>
  <summary>Please use the following BiocondaBot commands:</summary>

Everyone has access to the following BiocondaBot commands, which can be given in a comment:

<table>
  <tr>
    <td><code>@BiocondaBot please update</code></td>
    <td>Merge the master branch into a PR.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please add label</code></td>
    <td>Add the <code>please review & merge</code> label.</td>
  </tr>
  <tr>
    <td><code>@BiocondaBot please fetch artifacts</code></td>
    <td>Post links to CI-built packages/containers. <br />You can use this to test packages locally.</td>
  </tr>
</table>

Note that the <code>@BiocondaBot please merge</code> command is now depreciated. Please just squash and merge instead.

Also, the bot watches for comments from non-members that include `@bioconda/<team>` and will automatically re-post them to notify the addressed `<team>`.

</details>

### CODE REVIEW

File: recipes/r-archr/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+package:
+  name: r-archr
+  version: {{ version|replace("-", "_") }}
+
+source:
+  url: {{ github }}/archive/{{ commit }}.zip
+  sha256: aef7bd08d55494348cac8b9e0645a71a22a41dacbb7a7843cd939f1f31ee6d87
+
+build:
+  merge_build_host: True  # [win]
```
Comment (martin-g) [inline recipes/r-archr/meta.yaml:15]: Bioconda does not support Windows at the moment

File: recipes/r-archr/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+  run_exports:
+    - {{ pin_subpackage("r-archr", max_pin="x.x") }}
+
+requirements:
+  build:
+    - cross-r-base {{ r_base }}    # [build_platform != target_platform]
+    - {{ compiler('c') }}          # [not win]
+    - {{ compiler('m2w64_c') }}    # [win]
+    - {{ compiler('cxx') }}        # [not win]
+    - {{ compiler('m2w64_cxx') }}  # [win]
```
Comment (martin-g) [inline recipes/r-archr/meta.yaml:29]: - stdlib("c") ?

**Comment by mfansler** [comment]
Thanks for the feedback @martin-g! I will touch those up when I make other changes. I added some more thoughts to OP https://github.com/bioconda/bioconda-recipes/pull/51295#issue-2577914227, RE: versioning and upper bounding dependencies. I'm open to comments on that if you have time.

**Comment by jdblischak** [comment]
> okay if I list you as a maintainer here?

Yes, I agree to be a recipe maintainer

> Any thoughts about this also welcome!

I really don't like the idea of packaging the master or dev branches directly. How critical are these bug fixes? Could we create a bioconda for the latest GitHub release and then open an Issue to request another release?

The changes to master since the last release are minimal. Instead of pulling from a specific commit, we could apply a single patch.

https://github.com/GreenleafLab/ArchR/compare/v1.0.2...master

There are tons of changes on the dev branch. But who is using the dev branch?? The README in both the master and dev branches only instructs to install from master. And I'm pretty sure the pkgdown site is built from master.

https://github.com/GreenleafLab/ArchR/compare/v1.0.2...dev

Overall this package illustrates everything that is wrong with scientific software. It has established itself as critical for advancing research, yet it is being maintained as a side project by an overworked PI. We need more funding for Research Software Engineers! Fingers crossed that they can get funding to hire a maintainer.

https://github.com/GreenleafLab/ArchR/issues/2205#issuecomment-2337444214

**Comment by mfansler** [comment]
> *"Overall this package illustrates everything that is wrong with scientific software. It has established itself as critical for advancing research, yet it is being maintained as a side project by an overworked PI."*

This is partially what I hope we could address by generating stable builds on Bioconda. Users I know not only use the `dev` branch, but even resort to using some of the additional `dev_<blah>` branches at times because they provide piecemeal workarounds to specific issues. **The situation is a proper mess.** I agree that an upstream solution is ideal, but given that previous Issue threads indicate no imminent resolution, I think generating stable variants on Bioconda could help.

> *"Could we create a bioconda for the latest GitHub release and then open an Issue to request another release?"*

The alternative approach suggested would be building the stable Release and patching. My worry with that is it leads to maintainers here doing guesswork as to how things upstream *should* be resolved - a position I'm not eager to assume. Nevertheless, I'm open to that as *an initial recipe*. I.e., build from Release with a one patch for the threads fix. But I would also want to continue discussion (separate Issue) about variant builds for branches that have active use.

I've posted [an upstream request for Releases](https://github.com/GreenleafLab/ArchR/discussions/2215).

File: recipes/r-archr/meta.yaml (hunk 2 above)
Comment (mfansler) [reply recipes/r-archr/meta.yaml]: I think that is still only implemented on Conda Forge. Gave me [a linting error here](https://dev.azure.com/bioconda/bioconda-recipes/_build/results?buildId=71416&view=logs&j=eb1ca712-fec5-5d93-deb0-40983fe61f2f&t=c9ebc308-e058-58af-a4f4-866fa7649dbb&l=39).

**Comment by mfansler** [comment]
@BiocondaBot please fetch artifacts

**Review by mencian — APPROVED** [review]
LGTM, thanks for your input @jdblischak and @mfansler for doing the heavy lifting! Feel free merge if the binaries are working as intended.

**Pull request merged**